planner/funcdep/fd_graph.go

```diff
@@ -603,118 +603,119 @@ func (s *FDSet) MakeCartesianProduct(rhs *FDSet) {
-//			for d column determine different dependency.  NULL -> 1 and NULL -> NULL which breaks strict FD definition.
+//			<1> strict FD may be wakened as a lax one. But if at least one non-NULL column is part of the determinant, the
+//				strict FD can be preserved.
+//		        a  b  |  c     d     e
+//				------+----------------
+//			 	1  1  |  1    NULL   1
+//			    1  2  | NULL  NULL  NULL
+//			    2  1  | NULL  NULL  NULL
+//				left join with (a,b) * (c,d,e) on (a=c and b=1), if there is a strict FD {d} -> {e} on the rhs. After supplied
+//				with null values, {d} -> {e} are degraded to a lax one {d} ~~> {e} as you see. the origin and supplied null value
+//				for d column determine different dependency.  NULL -> 1 and NULL -> NULL which breaks strict FD definition.
 //
-//			If the determinant contains at least a not null column for example c here, FD like {c,d} -> {e} can survive
-//			after the left join. Because you can not find two same key, one from the origin rows and the other one from the
-//			supplied rows.
+//				If the determinant contains at least a not null column for example c here, FD like {c,d} -> {e} can survive
+//				after the left join. Because you can not find two same key, one from the origin rows and the other one from the
+//				supplied rows.
 //
-//			for lax FD, the supplied rows of null values don't affect lax FD itself. So we can keep it.
+//				for lax FD, the supplied rows of null values don't affect lax FD itself. So we can keep it.
 //
-//		<2> The FDSet should remove constant FD since null values may be substituted for some unmatched left rows. NULL is not a
-//			constant anymore.
+//			<2> The FDSet should remove constant FD since null values may be substituted for some unmatched left rows. NULL is not a
+//				constant anymore.
 //
-//      <3> equivalence FD should be removed since substituted null values are not equal to the other substituted null value.
+//	     <3> equivalence FD should be removed since substituted null values are not equal to the other substituted null value.
 //
 // 3: the newly added FD from filters should take some consideration as below:
 //
-//	 	<1> strict/lax FD: join key filter conditions can not produce new strict/lax FD yet (knowledge: 1&2).
+//		 	<1> strict/lax FD: join key filter conditions can not produce new strict/lax FD yet (knowledge: 1&2).
 //
-//		<2> constant FD from the join conditions is only used for checking other FD. We cannot keep itself.
-//			a   b  |  c     d
-//          -------+---------
-//          1   1  |  1     1
-//          1   2  | NULL NULL
-//          left join with (a,b) * (c,d) on (a=c and d=1), some rhs rows will be substituted with null values, and FD on rhs
-//          {d=1} are lost.
+//			<2> constant FD from the join conditions is only used for checking other FD. We cannot keep itself.
+//				a   b  |  c     d
```

Comment:
The header of this table is not correctly aligned.

Reply:
fix. So we need to add block comments for the table to make gofmt correctly aligned.